src/services/cronJobs/backupDonationImportJob.ts

```diff
@@ -0,0 +1,334 @@
+import config from '../../config';
+
+import { logger } from '../../utils/logger';
+import { schedule } from 'node-cron';
+import { detectAddressChainType } from '../../utils/networks';
+import {
+  createDonationQueryValidator,
+  validateWithJoiSchema,
+} from '../../utils/validators/graphqlQueryValidators';
+import { i18n, translationErrorMessagesKeys } from '../../utils/errorMessages';
+import { findProjectById } from '../../repositories/projectRepository';
+import { ProjStatus, Project } from '../../entities/project';
+import { Token } from '../../entities/token';
+import {
+  isTokenAcceptableForProject,
+  updateDonationPricesAndValues,
+} from '../donationService';
+import { findProjectRecipientAddressByNetworkId } from '../../repositories/projectAddressRepository';
+import {
+  findUserByWalletAddress,
+  setUserAsReferrer,
+} from '../../repositories/userRepository';
+import { ChainType } from '../../types/network';
+import { Donation } from '../../entities/donation';
+import { getChainvineReferralInfoForDonation } from '../chainvineReferralService';
+import { relatedActiveQfRoundForProject } from '../qfRoundService';
+import { NETWORK_IDS } from '../../provider';
+import { FetchedSavedFailDonationInterface } from '../../adapters/donationSaveBackup/DonationSaveBackupInterface';
+import { getDonationSaveBackupAdapter } from '../../adapters/adaptersFactory';
+
+const cronJobTime =
+  (config.get('DONATION_SAVE_BACKUP_CRONJOB_EXPRESSION') as string) ||
+  '0 0 * * 0';
+
+export const runSyncBackupServiceDonations = () => {
+  logger.debug('importBackupServiceDonations() has been called');
+  schedule(cronJobTime, async () => {
+    await importBackupServiceDonations();
+  });
+};
+
+// Minimum required params from the backup mongodb
+interface BackupDonationData {
+  _id: string;
+  chainId: number;
+  txHash: string;
+  amount: number;
+  token: {
+    address: number;
+  };
+  projectId: number;
+  anonymous: false;
+  nonce?: string;
+  walletAddress: string;
+  symbol: string;
+  chainvineReferred: string;
+  safeTransactionId?: string;
+}
+
+// Mock Mongo Methods to write a test
+export const importBackupServiceDonations = async () => {
+  const limit = 10;
+  let skip = 0;
+  let donations =
+    await getDonationSaveBackupAdapter().getNotImportedDonationsFromBackup({
+      limit,
+      skip,
+    });
+  while (donations.length > 0) {
+    for (const donation of donations) {
+      try {
+        await createBackupDonation(donation);
+        await getDonationSaveBackupAdapter().markDonationAsImported(
+          donation._id,
+        );
+      } catch (e) {
+        logger.error(`donation error with id ${donation._id}: `, e);
+        logger.error('donation error with params: ', donation);
```

Comment:
There is a point here, if importing an item gets an error, it will be fetched in the next round of importing again. So, it will endlessly be processed in each iteration.

Reply:
@aminlatifi 
How about adding another field(like isImported) for this kind of donations and set it true in this case, and not get them again

Reply:
Cool , I will do it

Reply:
I see the skip was removed. Could simply add a counter that increases when it enters the catch() statement. And on each iteration reset it to 0. 
But at the same time we could simply increase the limit and let the process run every hour or so. It shouldn't be a heavy process.
I was checking failed donations, and even on the previous round where we had big errors, the amount is never a huge number.